docs/hardhat-network/README.md

```diff
@@ -82,164 +85,39 @@ When a transaction fails without a reason, Hardhat Network will create a clear e
 
-### Mempool behavior
+More generally, Hardhat Network can be used to fork **any** network, not just mainnet. Even further, Hardhat Network can be used to fork **any EVM-compatible blockchain**, not just Ethereum.
 
-When automine is disabled, every sent transaction is added to the mempool, which contains all the transactions that could be mined in the future. Hardhat Network's mempool follows the same rules as geth. This means, among other things, that:
+There are other things you can do with a forked Hardhat Network. Check [our guide](../guides/mainnet-forking.md) to learn more.
 
-- Transactions with a higher gas price are included first
-- If two transactions can be included and both have the same gas price, the one that was received first is included first
-- If a transaction is invalid (for example, its nonce is lower than the nonce of the address that sent it), the transaction is dropped.
+### Mining modes
 
-You can get the list of pending transactions that will be included in the next block by using the "pending" block tag:
+Hardhat Network can be configured to **automine** blocks, immediately upon receiving each transaction, or it can be configured for **interval mining**, where a new block is mined periodically, incorporating as many pending transactions as possible.
 
-```js
-const pendingBlock = await network.provider.send("eth_getBlockByNumber", [
-  "pending",
-  false,
-]);
-```
-
-### Removing and replacing transactions
-
-Transactions in the mempool can be removed using the `hardhat_dropTransaction` method:
-
-```js
-const txHash = "0xabc...";
-await network.provider.send("hardhat_dropTransaction", [txHash]);
-```
-
-You can also replace a transaction by sending a new one with the same nonce as the one that it's already in the mempool but with a higher gas price. Keep in mind that, like in Geth, for this to work the new gas price has to be at least 10% higher than the gas price of the current transaction.
-
-### Configuring mining modes using RPC methods
-
-You can change the mining behavior on runtime using two RPC methods: `evm_setAutomine` and `evm_setIntervalMining`. For example, to disable automining:
+You can use one of these modes, both or neither. By default, only the automine mode is enabled.
 
-```js
-await network.provider.send("evm_setAutomine", [false]);
-```
+When automine is disabled, every sent transaction is added to the mempool, which contains all the transactions that could be mined in the future. Hardhat Network's mempool follows the same rules as geth. This means, among other things, that transactions are prioritized by gas price, then arrival time, and that invalid transactions are dropped. Pending transactions can be queried via the `eth_getBlockByNumber` RPC method (with `"pending"` as the block number argument), they can be removed using the `hardhat_dropTransaction` RPC method, and they can be replaced by a new transaction with the same nonce but a gas price 10+% higher.
```

Comment:
```suggestion
When automine is disabled, every sent transaction is added to the mempool, which contains all the transactions that could be mined in the future. Hardhat Network's mempool follows the same rules as geth. This means, among other things, that transactions are prioritized by gas/fee prices, then arrival time, and that invalid transactions are dropped. Pending transactions can be queried via the `eth_getBlockByNumber` RPC method (with `"pending"` as the block number argument), they can be removed using the `hardhat_dropTransaction` RPC method, and they can be replaced by a new transaction with the same nonce but a gas price 10+% higher.
```

London changed this

Reply:
Mmm, actually, the prioritization in London is much more complex. Txs are sorted by a new concept called "effective miner fee".

Maybe we can use "by the fees payed to the miner", whenever we are talking about sorting/priority?

The second part of the suggestion is valid. You need to increase both EIP-1559 fee params.

Reply:
Addressed in https://github.com/nomiclabs/hardhat/pull/1657/commits/8dbda2c4991fa3a592c288ffa0ae93e60ac21d80